tensorflow/python/ops/signal/fft_ops.py

```diff
@@ -399,11 +399,11 @@ def ifftshift(x, axes=None, name=None):
     x = _ops.convert_to_tensor(x)
     if axes is None:
       axes = tuple(range(x.shape.ndims))
-      shift = [-int(dim // 2) for dim in x.shape]
+      shift = [-_math_ops.cast(dim // 2, _dtypes.int32) for dim in _array_ops.shape(x)]
```

Comment:
Tensors are only iterable in eager mode, how about:
```
shift = -(_array_ops.shape(x) // 2)
```
(The cast is unnecessary because that was a trick to convert Dimension to an int, whereas tf.shape returns an int tensor already, though we should keep the order of operations)

Reply:
True, plus it wasn't going to work anyway with a `None`.